packages/mail-client/src/mail-client.ts

```diff
@@ -217,27 +225,36 @@ async function sendMailInParallel<T extends MailConfig>(
   mailConfigs: T[]
 ): Promise<MailResponse[]> {
   const promises: Promise<MailResponse>[] = mailConfigs.map(
-    (mailConfig, mailConfigIndex, mailConfigsArray) => {
-      logger.debug(
-        `Sending email ${mailConfigIndex + 1}/${mailConfigsArray.length}...`
-      );
-      return transport.sendMail({
-        ...mailConfigsFromDestination,
-        ...mailConfig
-      });
+    async (mailConfig, mailConfigIndex, mailConfigsArray) => {
+      try {
+        logger.debug(
+          `Sending email ${mailConfigIndex + 1}/${mailConfigsArray.length}...`
+        );
+        const response = await transport.sendMail({
+          ...mailConfigsFromDestination,
+          ...mailConfig
+        });
+        logger.debug(
+          `...email ${mailConfigIndex + 1}/${mailConfigs.length} with subject "${
+            mailConfig.subject
+          }" was sent successfully.`
+        );
+        return response;
+      } catch (error) {
+        logger.error(
+          `Failed to send email ${mailConfigIndex + 1}/${mailConfigs.length} with subject "${
+            mailConfig.subject
+          }". Error: ${error.message}`
+        );
+        return {
+          response: error.response,
+          rejected: error.rejected
+        };
+      }
     }
   );
 
-  return Promise.all(promises).then(responses => {
-    responses.forEach((_, responseIndex) =>
-      logger.debug(
-        `...email ${responseIndex + 1}/${mailConfigs.length} for subject "${
-          mailConfigs[responseIndex].subject
-        }" was sent successfully`
-      )
-    );
-    return responses;
-  });
+  return Promise.all(promises);
```

Comment:
[q] The reject error we could get here is always only the first error right?

Reply:
No, this function will not reject with only the first error (or any unhandled error) because each individual error is handled in the catch and returns a MailResponse type
```
{
   response: error.response,
   rejected: error.rejected
}
```

Reply:
Actually I should write a test for parallel case as well.